app/jobs/solar/base_solar_loader_job.rb

```diff
@@ -0,0 +1,47 @@
+module Solar
+  #Base class for all the jobs that load solar data
+  class BaseSolarLoaderJob < ApplicationJob
+    include Rails.application.routes.url_helpers
+
+    queue_as :default
+
+    def priority
+      5
+    end
+
+    def perform(installation:, notify_email:, start_date: nil, end_date: nil)
+      @installation = installation
+      @upserter = upserter(start_date, end_date)
+      @upserter.perform
+      send_notification(notify_email)
+    rescue => e
+      Rollbar.error(e, job: :import_solar_edge_readings)
+      send_failure_notification(notify_email, e)
+      false
+    end
+
```

Comment:
I'd be tempted to do add something like this for upserter and solar_feed_type methods to ensure they are implemented in subclasses:

  ```
  private

  def upserter(start_date, end_date)
    raise "Not implemented"
  end

  def solar_feed_type
    raise "Not implemented"
  end
```

Reply:
Good call. I've never been sure about the best way to do this in Ruby, but having some kind of stub makes sense.

Reply:
Looks good @ldodds 👍